crates/bevy_render/src/extract_component.rs

```diff
@@ -28,6 +28,24 @@ impl<C: Component> DynamicUniformIndex<C> {
     }
 }
 
+impl<C: Component> Default for DynamicUniformIndex<C> {
+    fn default() -> Self {
+        Self {
+            index: 0,
+            marker: PhantomData,
+        }
+    }
+}
+
+impl<C: Component> Clone for DynamicUniformIndex<C> {
+    fn clone(&self) -> Self {
+        Self {
+            index: self.index,
+            marker: PhantomData,
+        }
+    }
+}
```

Comment:
Is the manual `Clone` necessary here?
It doesn't relax the `C: Component` bound and `Copy` is still derived.

Reply:
`PhantomData<T>` only implements `Clone` iff `T: Clone`, which also transitively holds for the derived impl. This implements `Clone` and `Default` regardless of what `T` is.